lib/pg_eventstore/subscriptions/events_processor.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module PgEventstore
+  # This class actually processes events.
+  # @!visibility private
+  class EventsProcessor
+    include Extensions::CallbacksExtension
+    extend Forwardable
+
+    def_delegators :@basic_runner, :state, :start, :stop, :wait_for_finish, :stop_async, :restore, :running?
+
+    # @param handler [#call]
+    def initialize(handler)
+      @handler = handler
+      @raw_events = []
+      @basic_runner = BasicRunner.new(0, 5)
+      attach_runner_callbacks
+    end
+
+    # @param raw_events [Array<Hash>]
+    # @return [void]
+    def feed(raw_events)
+      callbacks.run_callbacks(:feed, raw_events.last&.dig('global_position'))
+      @raw_events.push(*raw_events)
+    end
+
+    # Number of unprocessed events which are currently in a queue
+    # @return [Integer]
+    def events_left_in_chunk
+      @raw_events.size
+    end
+
+    private
+
+    # @param raw_event [Hash]
+    # @return [void]
+    def process_event(raw_event)
+      callbacks.run_callbacks(:process, raw_event['global_position']) do
+        @handler.call(raw_event)
+      end
+    end
+
+    def attach_runner_callbacks
+      @basic_runner.define_callback(:process_async, :before, method(:process_async))
+      @basic_runner.define_callback(:after_runner_died, :before, method(:after_runner_died))
+      @basic_runner.define_callback(:before_runner_restored, :before, method(:before_runner_restored))
+      @basic_runner.define_callback(:change_state, :before, method(:change_state))
+    end
+
+    def process_async
+      raw_event = @raw_events.shift
+      return sleep 0.5 if raw_event.nil?
+
+      process_event(raw_event)
+    rescue
```

Comment:
Default behaviour of `rescue` is to rescue from `StandardError`. So that line is equal to 

```ruby
rescue StandardError
```

If you meant that - then yes. I would like to catch every exception which can happen inside event handler